tests/io/test_utils.py

```diff
@@ -16,3 +18,19 @@ def test_get_sql_metadata():
     assert metadata2["query"] == query
     assert metadata2["rows"] == 5
     assert metadata3["rows"] is None
+
+
+def test_get_file_metadata(tmp_path: pathlib.Path):
+    file_path = tmp_path / "test.txt"
+    file_path.write_text("test")
+    metadata = get_file_metadata(file_path)
+    assert metadata["file_metadata"]["path"] == str(file_path)
+    assert metadata["file_metadata"]["size"] > 0
+    assert metadata["file_metadata"]["last_modified"] == file_path.stat().st_mtime
+    assert metadata["file_metadata"]["timestamp"] is not None
+
+
+def test_get_file_metadata_url_schema():
+    url = "s3://bucket/key"
+    metadata = get_file_metadata(url)
+    assert metadata["file_metadata"]["path"] == url
```

Comment:
shouldn't this check scheme?

Reply:
schema yes, not notes cause that's not really part of the API